cmd/soroban-rpc/lib/preflight/src/lib.rs

```diff
@@ -664,3 +668,32 @@ fn extract_error_string<T>(simulation_result: &Result<T>, go_storage: &GoLedgerS
         }
     }
 }
+
+// xdr_to_json takes in a string name of an XDR type in the Stellar Protocol
+// (i.e. from the stellar_xdr crate) as well as a raw byte structure and returns a
+// JSONified string of said structure.
+#[no_mangle]
+pub extern "C" fn xdr_to_json(typename: *mut libc::c_char, xdr: CXDR) -> *mut libc::c_char {
+    let type_str = from_c_string(typename);
+    let the_type = match xdr::TypeVariant::from_str(&type_str) {
+        Ok(t) => t,
+        Err(e) => {
+            return string_to_c(format!(r#"{{ "error": "{e}", "type": "{type_str}" }}"#));
```

Comment:
Is this preferable or should I do a similar thing to preflight where we catch and unwind panics? I feel like this is pretty clean but cc @2opremio for opinions

Reply:
I did the catch and unwind to incorporate @leighmcculloch's suggestion to avoid the JSON round-trip :+1: